astropy/io/misc/yaml.py

```diff
@@ -0,0 +1,295 @@
+    z: !astropy.units.Quantity
+      unit: *id001
+      value: 3000.0
+  out_subfmt: '*'
+  precision: 3
+  scale: utc
+
+  >>> ty = yaml.load(td)
+  >>> ty
+  <Time object: scale='utc' format='mjd' value=2457389.0>
+
+  >>> ty.location
+  <EarthLocation (1000.0, 2000.0, 3000.0) km>
+"""
+
+from __future__ import absolute_import
+
+import base64
+import numpy as np
+
+from ...time import Time, TimeDelta
+from ... import units as u
+from ... import coordinates as coords
+from ...utils.data_info import _get_obj_attrs_map
+
+try:
+    import yaml
+except ImportError:
+    raise ImportError('`import yaml` failed, PyYAML package is required for YAML')
+
+
+__all__ = ['AstropyLoader', 'AstropyDumper', 'load', 'load_all', 'dump']
+
+def _unit_representer(dumper, obj):
+    out = {'name': str(obj.to_string())}
+    return dumper.represent_mapping(u'!astropy.units.Unit', out)
+
+
+def _unit_constructor(loader, node):
+    map = loader.construct_mapping(node)
+    return u.Unit(map['name'])
+
+
+def _time_representer(dumper, obj):
+    out = obj.info._represent_as_dict()
+    return dumper.represent_mapping(u'!astropy.time.Time', out)
+
+
+def _time_constructor(loader, node):
+    map = loader.construct_mapping(node)
+    out = Time.info._construct_from_dict(map)
+    return out
+
+
+def _timedelta_representer(dumper, obj):
+    out = obj.info._represent_as_dict()
+    return dumper.represent_mapping(u'!astropy.time.TimeDelta', out)
+
+
+def _timedelta_constructor(loader, node):
+    map = loader.construct_mapping(node)
+    out = TimeDelta.info._construct_from_dict(map)
+    return out
+
+
+def _ndarray_representer(dumper, obj):
+    if obj.flags['C_CONTIGUOUS']:
+        obj_data = obj.data
+    else:
+        cont_obj = np.ascontiguousarray(obj)
+        assert(cont_obj.flags['C_CONTIGUOUS'])
+        obj_data = cont_obj.data
+    data_b64 = base64.b64encode(bytes(obj_data))
+    out = dict(__ndarray__=data_b64,
+               dtype=str(obj.dtype),
+               shape=obj.shape)
+
+    return dumper.represent_mapping(u'!numpy.ndarray', out)
+
+
+def _ndarray_constructor(loader, node):
+    map = loader.construct_mapping(node)
+    data = base64.b64decode(map['__ndarray__'])
+    return np.frombuffer(data, map['dtype']).reshape(map['shape'])
+
+
+def _quantity_representer(tag):
+    def representer(dumper, obj):
+        out = obj.info._represent_as_dict()
+        return dumper.represent_mapping(tag, out)
+    return representer
+
+
+def _quantity_constructor(cls):
+    def constructor(loader, node):
+        map = loader.construct_mapping(node)
+        return cls.info._construct_from_dict(map)
+    return constructor
+
+
+def _earthlocation_representer(dumper, obj):
+    out = _get_obj_attrs_map(obj, ('x', 'y', 'z', 'ellipsoid'))
+    return dumper.represent_mapping(u'!astropy.coordinates.earth.EarthLocation', out)
+
+
+def _earthlocation_constructor(loader, node):
+    map = loader.construct_mapping(node)
+    ellipsoid = map.pop('ellipsoid')
```

Comment:
I had not realized that `EarthLocation` is a `Quantity` subclass.  Giving it its own `info` in 3f778c1 allowed making it just another one of the list of `Quantity` subclasses, nice!